api/scpca_portal/models/library.py

```diff
@@ -34,3 +37,38 @@ class Modalities:
     modality = models.TextField(choices=Modalities.CHOICES)
     scpca_id = models.TextField(unique=True)
     workflow_version = models.TextField()
+
+    @classmethod
+    def get_from_dict(cls, data, formats):
+        library = cls(
+            formats=formats,
+            is_multiplexed=("demux_samples" in data),
+            metadata=data,
+            modality=(
+                Library.Modalities.SINGLE_CELL
+                if "filtered_cell_count" in data
+                else Library.Modalities.SPATIAL
+            ),
+            scpca_id=data["scpca_library_id"],
+            workflow_version=data["workflow_version"],
+        )
+
+        return library
+
+    @classmethod
+    def bulk_create_from_dicts(cls, library_jsons: List[Dict], sample, sample_dir: Path) -> None:
```

Comment:
I think we should have a discussion about this. There is a method on `Project` called `get_sample_input_data_dir` which takes in a `sample_id` as an argument and returns the Path object for the `sample_dir`. Part of me feels like we should just move this method to a property method on `Sample`, but there are occurences where this `sample_dir` is needed before the creation of `Sample` objects, so that's a question needed to be addressed. 

As it pertains to your question here, I was able to get around passing through the `sample_dir` into `Libary::bulk_create_from_dicts`. Please see the group of most recent commits.